pkg/services/sqlstore/migrations/accesscontrol/dashboard_permissions.go

```diff
@@ -661,6 +661,170 @@ func (m *managedFolderLibraryPanelActionsMigrator) Exec(sess *xorm.Session, mg *
 	return nil
 }
 
+const ManagedDashboardAnnotationActionsMigratorID = "managed dashboard permissions annotation actions migration"
+
+func AddManagedDashboardAnnotationActionsMigration(mg *migrator.Migrator) {
+	mg.AddMigration(ManagedDashboardAnnotationActionsMigratorID, &managedDashboardAnnotationActionsMigrator{})
+}
+
+type managedDashboardAnnotationActionsMigrator struct {
+	migrator.MigrationBase
+}
+
+func (m *managedDashboardAnnotationActionsMigrator) SQL(dialect migrator.Dialect) string {
+	return CodeMigrationSQL
+}
+
+func (m *managedDashboardAnnotationActionsMigrator) Exec(sess *xorm.Session, mg *migrator.Migrator) error {
+	// Check if roles have been populated and return early if they haven't - this avoids logging a warning from hasDefaultAnnotationPermissions
+	roleCount, err := sess.Where(`uid IN ("basic_viewer", "basic_editor", "basic_admin")`).Count(&ac.Role{})
+	if err != nil {
+		return fmt.Errorf("failed to check if basic roles have been populated: %w", err)
+	}
+	// Role count will be 0 either for new Grafana installations (in that case no managed roles will exist either, and the next conditional will return nil)
+	// or for OSS instances, for which basic role permissions can't be changed, so we don't need to run the default permission check in that case.
+	if roleCount != 0 {
```

Comment:
Sadly, if I'm not mistaken, this does not work, the roles are created even for OSS by `builtinRoleMigrator` but they don't have any permissions.

Reply:
Ooh, I see. Man, seeding is complicated 😆.